File: fmcg-android/app/src/main/java/com/fmcg/app/domain/model/Store.kt
package com.fmcg.app.domain.model

data class Store(
    val id: Int,
    val name: String,
    val ownerName: String?,
    val phone: String?,
    val address: String?,
    val latitude: Double,
    val longitude: Double,
    val notes: String?,
    val photoUrl: String?,
)

/** Fields a rep supplies when creating or editing a store. */
data class StoreDraft(
    val name: String,
    val ownerName: String?,
    val phone: String?,
    val address: String?,
    val latitude: Double,
    val longitude: Double,
    val notes: String?,
)

File: fmcg-backend/app/models/store.py
"""Store (customer) model. Coordinates stored as doubles."""
from sqlalchemy import Float, ForeignKey, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base
from app.models.mixins import TimestampMixin


class Store(TimestampMixin, Base):
    __tablename__ = "stores"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(160), index=True)
    owner_name: Mapped[str | None] = mapped_column(String(120), nullable=True)
    phone: Mapped[str | None] = mapped_column(String(20), nullable=True)
    address: Mapped[str | None] = mapped_column(String(300), nullable=True)
    latitude: Mapped[float] = mapped_column(Float)
    longitude: Mapped[float] = mapped_column(Float)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    photo_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    created_by: Mapped[int | None] = mapped_column(
        ForeignKey("users.id", ondelete="SET NULL"), nullable=True
    )
